manifests/plugin/container.pp

```diff
@@ -9,6 +9,11 @@
   String $location_prefix = '/pulpcore_registry',
   String $registry_version_path = '/v2/',
 ) {
+  # This is like pulpcore::apache's value, but slightly different
+  $api_default_request_headers = [
```

Comment:
Do we really need this?

https://httpd.apache.org/docs/current/mod/mod_headers.html#requestheader says

set
    The request header is set, replacing any previous header with this name

So while the content of the file is changed, the behavior should not?

(Unless the cn map is empty)

Reply:
The CN map contains "if" expressions. My understanding is the first line 100% certain clears it. Then IF the cert's CN equals something, it sets the header.

Reply:
Right, so the flow is:
* use `unset` to remove *any* header by that name, wherever it comes from
* use `set` to set it to the CN (if there is a CN!)
* use `set` to set it to `admin` (if the expression matches, resetting a possibly previously set CN-based name)

I honestly see no reason (ever) to set the CN-based remote user, as such user just doesn't exist in Pulp (in our deployment)

Reply:
It was added in ~~https://github.com/theforeman/puppet-pulpcore/pull/186~~ https://github.com/theforeman/puppet-pulpcore/pull/105, but I don't really see a reason for it.

Reply:
My vote: don't add this complexity (same variable name, different values) back in, it's confusing as hell.
Do the releases we need to do.

Drop CN-stuff in an 11.x

Reply:
Should we then include 11.x in the upcoming releases? I'd actually feel more comfortable if we only had the explicit unset and explicit mapping.

Reply:
I'm not sure…

Reply:
fwiw, I tested the current (two-`set`) config with https://github.com/theforeman/forklift/pull/1853 and it pushes containers fine, so :woman_shrugging:

Reply:
I'm leaning to merging this now as is because it keeps it minimal and then resolve it properly in the next release cycle.